gateway/utils/syncman/store_sc.go

```diff
@@ -0,0 +1,171 @@
+package syncman
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/sirupsen/logrus"
+	sc "github.com/spaceuptech/space-api-go"
+	"github.com/spaceuptech/space-api-go/db"
+	"github.com/spaceuptech/space-api-go/types"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/spaceuptech/space-cloud/gateway/config"
+)
+
+type SCStore struct {
+	clusterID  string
+	collection string
+	scDatabase *db.DB
+	kube       *kubernetes.Clientset
+}
+
+type scTableScheam struct {
+	clusterId string `json:"cluster_id"`
+	ProjectId string `json:"project_id"`
+	Project   string `json:"project"`
+}
+
+func NewSCStore(clusterID, scStoreProjectName, scStoreAddr, scStoreDatabaseName string) (*SCStore, error) {
+	// Create the kubernetes client
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	kube, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SCStore{
+		clusterID:  clusterID,
+		kube:       kube,
+		scDatabase: sc.New(scStoreProjectName, scStoreAddr, false).DB(scStoreDatabaseName)}, nil
+}
+
+func (s *SCStore) Register() {
+	// kubernetes will handle this automatically
+}
+
+func (s *SCStore) WatchProjects(cb func(projects []*config.Project)) error {
+	subObj := s.scDatabase.LiveQuery(s.collection).Where(types.M{"cluster_id": s.clusterID}).Subscribe()
+	for value := range subObj.C() {
+		if err := value.Err(); err != nil {
+			logrus.Errorf("error watching projects in sc store - %v", err)
+			continue
+		}
+		projects, err := s.getProjects(subObj.GetSnapshot())
+		if err != nil {
+			logrus.Errorf("error watching projects in sc store unable to get projects - %v", err)
+			continue
+		}
+		cb(projects)
+	}
+	return nil
+}
```

Comment:
Let's watch on ConfigMap instead of doing a SC live query. Lets store all our data in Config Maps